llvm/lib/Target/DirectX/DXILPostOptimizationValidation.cpp

```diff
@@ -214,13 +227,20 @@ static void validateRootSignature(Module &M,
     Builder.trackBinding(dxil::ResourceClass::Sampler, S.RegisterSpace,
                          S.ShaderRegister, S.ShaderRegister, &S);
 
-  Builder.calculateBindingInfo(
+  hlsl::BindingInfo Info = Builder.calculateBindingInfo(
       [&M](const llvm::hlsl::BindingInfoBuilder &Builder,
            const llvm::hlsl::BindingInfoBuilder::Binding &ReportedBinding) {
         const llvm::hlsl::BindingInfoBuilder::Binding &Overlaping =
             Builder.findOverlapping(ReportedBinding);
         reportOverlappingRegisters(M, ReportedBinding, Overlaping);
       });
+  for (const ResourceInfo &RI : DRM) {
+    const ResourceInfo::ResourceBinding &Binding = RI.getBinding();
+    ResourceClass RC = DRTM[RI.getHandleTy()].getResourceClass();
+    if (!Info.isBound(RC, Binding.Space,
+                      {Binding.LowerBound, Binding.LowerBound + Binding.Size}))
```

Comment:
```suggestion
                      {Binding.LowerBound, Binding.LowerBound + Binding.Size - 1}))
```

Just a quick check that the size is inclusive or exclusive and if we have test case to demonstrate

Reply:
Found a bug, fixed it, added test